### PR TITLE
Feature: Add login page #8

### LINKED ISSUE #8: Create login page

### PR DESCRIPTION
**Pull request by bdom5ki**
Closes #8

### CODE REVIEW

**Comment by wjureczka** [comment]
I would like to introduce you to BEM methodology: http://getbem.com/
Let's try to refactor class names according to BEM :)

File: packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  align-items: center;
+  border-radius: 2px;
+  background: white;
+  width: 500px;
+  padding: 30px 100px;
+  box-sizing: border-box;
+  -webkit-box-shadow: 0 0 12px 4px rgba(0, 0, 0, 0.1);
+  box-shadow: 0 0 12px 4px rgba(0, 0, 0, 0.1);
+
+  > * {
```
Comment (wjureczka) [inline packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.scss]: Please avoid using such selectors

File: packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.scss [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  .header {
+    margin-bottom: 20px;
+    font-size: 1.6rem;
+    text-align: center;
+    opacity: .5;
+  }
+}
+
+:host ::ng-deep p {
```
Comment (wjureczka) [inline packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.scss]: Is it possible to style button and paragraph without using :host and ::ng-deep? Did you try?

Comment (wjureczka) [reply packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.scss]: https://dev.to/jwp/angular-10-ng-deep-and-host-pseudo-classes-1kbg

File: packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+
+@Component({
+  selector: 'app-credentials-form',
+  templateUrl: './credentials-form.component.html',
+  styleUrls: ['./credentials-form.component.scss']
+})
+export class CredentialsFormComponent implements OnInit {
+
+  form: FormGroup;
```
Comment (wjureczka) [inline packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.ts:11]: Let's make them private, wdyt?

Comment (bdom5ki) [reply packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.ts:11]: I use it as formGroup in html so private throws compilation error :(

File: packages/frontend/src/app/views/sessions/login/login.component.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
+:host {
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.footer div {
```
Comment (wjureczka) [inline packages/frontend/src/app/views/sessions/login/login.component.scss]: Let's make it bemish

Comment (wjureczka) [reply packages/frontend/src/app/views/sessions/login/login.component.scss]: The other solution is to use:
display: flex;
align-items: center;
justify-content: space-between;

File: packages/frontend/src/styles.scss [hunk 5]
```diff
@@ -9,3 +9,16 @@ body {
...
+a {
+  cursor: pointer;
+  text-decoration: none;
+}
+
+a:hover {
+  text-decoration: underline;
+}
+
+form .mat-form-field-underline {
```
Comment (wjureczka) [inline packages/frontend/src/styles.scss]: Let's put it where form component is

**Comment by wjureczka** [comment]
https://angular.io/guide/styleguide#overall-structural-guidelines
I would recommend to follow angular style guide and separate login-page to its own folder and module.

**Comment by wjureczka** [comment]
Overall nice PR!
Random reddit meme:
image

File: packages/frontend/src/styles.scss (hunk 5 above)
Comment (bdom5ki) [reply packages/frontend/src/styles.scss]: I think it's won't work  for mat-form-field-underline from component styles with component default encapsulation. However I'll check to make sure. Or maybe there are other solutions?

File: packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.scss (hunk 2 above)
Comment (bdom5ki) [reply packages/frontend/src/app/shared/forms/credentials-form/credentials-form.component.scss]: Ok, I'll try something else

**Review by wjureczka — DISMISSED** [review]

**Review by wjureczka — APPROVED** [review]

**Pull request merged**